graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -258,6 +267,220 @@ public int maxBatchSize() {
             return flattener.results(mapper, key.getQuery().getType(), Optional.ofNullable(key.getQuery().getLimit()));
         });
     }
+    
+    @Override
+    public <T extends Table> CompletableFuture<List<T>> queryHistory(DatabaseQueryHistoryKey<T> key) {
```

Comment:
would try split this into some different methods. Maybe one for each of the outer if statements. Is so long.